extensions/claude/src/manager/claude-inference-manager.ts

```
@@ -0,0 +1,207 @@
+    if (!apiKey || typeof apiKey !== 'string') throw new Error('invalid apiKey');
+
+    // save the key in secret storage
+    await this.saveToken(apiKey);
+
+    // use dedicated method to register connection
+    await this.registerInferenceProviderConnection({ token: apiKey });
+  }
+
+  dispose(): void {
+    this.provider?.dispose();
+    this.connections.forEach(disposable => disposable.dispose());
```

Biome (2.4.10)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)